### PR TITLE
Adding report for stale keys

### PR DESCRIPTION
**Pull request by cloudnoize**
used for testing pruning

### CODE REVIEW

File: kvbc/include/kvbc_adapter/replica_adapter.hpp [hunk 1]
```diff
@@ -223,6 +223,9 @@ class ReplicaBlockchain : public IBlocksDeleter,
   const concord::kvbc::categorization::KeyValueBlockchain *const getReadOnlyCategorizedBlockchain() {
     return kvbc_.get();
   }
+  std::map<std::string, std::vector<std::string>> getBlockStaleKeys(BlockId block_id) const {
```
Comment (arc-vmware) [inline kvbc/include/kvbc_adapter/replica_adapter.hpp:226]: nit: These functions are in v1 also. Should we encapsulate these in an interface (IPrunerMetric) and then add them to v1, v4, and adapter.

File: kvbc/include/kvbc_adapter/replica_adapter.hpp [hunk 2]
```diff
@@ -223,6 +223,9 @@ class ReplicaBlockchain : public IBlocksDeleter,
...
+    return kvbc_ == nullptr ? v4_kvbc_->getBlockStaleKeys(block_id) : kvbc_->getBlockStaleKeys(block_id);
```
Comment (arc-vmware) [inline kvbc/include/kvbc_adapter/replica_adapter.hpp:227]: This check can be avoided at the adapter layer if we use an interface. In the long run interface will help when there will be many implementations.

**Review by arc-vmware — APPROVED** [review]
Thanks @cloudnoize , great code. I have some small comments, approving to avoid another cycle in code review.

File: kvbc/include/kvbc_adapter/replica_adapter.hpp (hunk 2 above)
Comment (cloudnoize) [reply kvbc/include/kvbc_adapter/replica_adapter.hpp:227]: you're right, but this functionality is so insignificant, I don't think it worth the change now

**Pull request merged**